### PR TITLE
fix(list-item): Improve item toggle behavior

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7017

## Summary

- Honors `event.defaultPrevented` for click/key events so users can prevent them from doing their thing.
- Will no longer toggle selection when selection-mode is set to "none".
- Changes key from toggling selection from space to enter.

### CODE REVIEW

File: src/components/list-item/list-item.tsx [hunk 1]
```diff
@@ -604,28 +605,35 @@ export class ListItem
...
     const composedPath = event.composedPath();
-    const { containerEl, contentEl, actionsStartEl, actionsEndEl, open, openable } = this;
+    const { containerEl, contentEl, actionsStartEl, actionsEndEl, open, openable, selectionMode } =
+      this;
 
     const cells = [actionsStartEl, contentEl, actionsEndEl].filter(Boolean);
     const currentIndex = cells.findIndex((cell) => composedPath.includes(cell));
 
-    if (key === " ") {
+    if (isActivationKey(key) && selectionMode && selectionMode !== "none") {
```
Comment (driskull) [inline src/components/list-item/list-item.tsx:644]: `treegrid` role says enter should perform default row action and not space. cc @geospatialem what do you think? Should we just honor `Enter` key like the spec defines? 

https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/

File: src/components/list-item/list-item.tsx [hunk 2]
```diff
@@ -604,28 +605,35 @@ export class ListItem
...
+      this.selected = !selected;
+    } else if (selectionMode === "single-persist") {
       this.selected = true;
     }
 
     this.calciteListItemSelect.emit();
   };
 
   handleItemKeyDown = (event: KeyboardEvent): void => {
+    if (event.defaultPrevented) {
```
Comment (driskull) [inline src/components/list-item/list-item.tsx:632]: We can honor users preventing default for keydown.

File: src/components/list-item/list-item.tsx (hunk 1 above)
Comment (geospatialem) [reply src/components/list-item/list-item.tsx:644]: Yes, concur with changing to `Enter` from `space` for alignment, and also seems more intuitive. Once updated, can update the docs site to reflect the change (looks like `space` was not previously doc'd).

Comment (driskull) [reply src/components/list-item/list-item.tsx:644]: @geospatialem should the "default action" be to toggle selection or toggle open state of children?

Comment (geospatialem) [reply src/components/list-item/list-item.tsx:644]: Was the children open state previously handled by `enter`? Thought it was through the arrow keys? Would assume the default would be toggle selection.

**Review by jcfranco — APPROVED** [review]
💥

Could you add a test before merging?

Also, can you update the PR title to reflect what the improvement was?

**Comment by jcfranco** [comment]
Would "improve item toggle behavior" or similar work?

**Pull request merged**